File: apps/java/src/main/java/com/functions/HealthFunction.java
package com.functions;

import com.microsoft.azure.functions.*;
import com.microsoft.azure.functions.annotation.*;
import java.time.Instant;
import java.util.Optional;

/**
 * Health check endpoint — returns app liveness status.
 */
public class HealthFunction {

    @FunctionName("health")
    public HttpResponseMessage run(
            @HttpTrigger(
                name = "req",
                methods = {HttpMethod.GET},
                authLevel = AuthorizationLevel.ANONYMOUS,
                route = "health")
            HttpRequestMessage<Optional<String>> request,
            final ExecutionContext context) {

        context.getLogger().info("Health check invoked");

        String body = String.format(
            "{\"status\":\"healthy\",\"timestamp\":\"%s\",\"version\":\"1.0.0\"}",
            Instant.now().toString());

        return request.createResponseBuilder(HttpStatus.OK)
                .header("Content-Type", "application/json")
                .body(body)
                .build();
    }
}

File: apps/dotnet/Functions/HealthFunction.cs
using System.Text.Json;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.Functions.Worker;
using Microsoft.Extensions.Logging;

namespace AzureFunctionsGuide.Functions;

/// <summary>
/// Health check endpoint — returns app liveness status.
/// Route: GET /api/health
/// </summary>
public class HealthFunction
{
    private readonly ILogger<HealthFunction> _logger;

    public HealthFunction(ILogger<HealthFunction> logger)
    {
        _logger = logger;
    }

    [Function("health")]
    public IActionResult Run(
        [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "health")] HttpRequest req)
    {
        _logger.LogInformation("Health check invoked");

        var result = new
        {
            status = "healthy",
            timestamp = DateTime.UtcNow.ToString("o"),
            version = "1.0.0"
        };

        return new OkObjectResult(result);
    }
}
